src/shared/lib/hooks/useOnFocus.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useRef } from 'react';
+
+import { useIsFocused } from '@react-navigation/native';
+
+const useOnFocus = (callback: () => void) => {
+  const isFocused = useIsFocused();
+
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
+
+  useEffect(() => {
+    if (isFocused) {
+      callbackRef.current();
+    }
+  }, [isFocused]);
```

Comment:
Please use useFocusEffect instead as it serves exactly the purpose we want to utilize. Here is a quick comparison:
https://www.dhiwise.com/post/how-to-use-the-usefocuseffect-hook-in-react-effectively#comparing-usefocuseffect-and-useisfocused-in-react-native

Reply:
ok